hail/src/main/scala/is/hail/sparkextras/MultiWayJoinRDD.scala

```diff
@@ -0,0 +1,52 @@
+package is.hail.sparkextras
+
+import org.apache.spark.{OneToOneDependency, Partition, SparkContext, TaskContext}
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+object MultiWayJoinRDD {
+  def apply[T: ClassTag , V: ClassTag](
+    rdds: IndexedSeq[RDD[T]],
+    preservesPartitioning: Boolean = false
+  )(f: (Array[Iterator[T]]) => Iterator[V]): MultiWayJoinRDD[T, V] = {
+    new MultiWayJoinRDD(rdds.head.sparkContext, rdds, f, preservesPartitioning)
+  }
+}
+
+private class MultiWayJoinPartition[T: ClassTag](
+  idx: Int,
+  @transient private val rdds: Seq[RDD[T]]
+) extends Partition {
+  override val index: Int = idx
+  var partitionVals = rdds.map(rdd => rdd.partitions(idx))
+  def partitions = partitionVals
+}
+
+class MultiWayJoinRDD[T: ClassTag, V: ClassTag](
+  sc: SparkContext,
+  var rdds: IndexedSeq[RDD[T]],
+  var f: (Array[Iterator[T]]) => Iterator[V],
+  preservesPartitioning: Boolean = false
+) extends RDD[V](sc, rdds.map(x => new OneToOneDependency(x))) {
+
+  override val partitioner = if (preservesPartitioning) firstParent[Any].partitioner else None
```

Comment:
style: firstParent[_]

Reply:
I take option 'consistency' below, the suggestion made fails with `unbound wildcard type`.

Reply:
ah, interesting. fair enough